### PR TITLE
Re-add boosters

### PR DESCRIPTION
**Pull request by Gymnasiast**
screenshot

### CODE REVIEW

**Comment by IntelOrca** [comment]
How to do you place them? They don't show up on the special list for me.

**Comment by Gymnasiast** [comment]
Which track type did you try? I have added them for the looping and corkscrew RC (like in RCT1) and the Twister, Giga and Junior RC (RCTC), plus the vertical RC, since that keeps parity with the Twister.

**Comment by IntelOrca** [comment]
All of them

**Comment by Gymnasiast** [comment]
diamond heights 2017-01-01 20-15-01

**Comment by AaronVanGeffen** [comment]
What about the graphics for booster tracks? This PR doesn't seem to add any, and I believe RCT2 does not include them — does it?

**Comment by IntelOrca** [comment]
Okay, they work sometimes - like for Juinior RC.

If you open 'Build you own Six Flags park' and try building the corkscrew coaster, it won't have them.
I also have a saved game which doesn't have them for giga coaster, corkscrew or looping.

Maybe its due to the available track configurations for a vehicle...?

[all_rides_unlocked.sv6.txt](https://github.com/OpenRCT2/OpenRCT2/files/680147/all_rides_unlocked.sv6.txt)

**Comment by Gymnasiast** [comment]
Oh, that will be it. I have the game set to select-by-track-type, which doesn't suffer from that.

File: src/object/RideObject.cpp [hunk 1]
```diff
@@ -20,6 +20,7 @@
 #include "../core/Util.hpp"
 #include "ObjectRepository.h"
 #include "RideObject.h"
+#include "../ride/track.h"
```
Comment (IntelOrca) [inline src/object/RideObject.cpp]: Sort the includes

File: src/object/RideObject.cpp [hunk 2]
```diff
@@ -79,6 +80,21 @@ void RideObject::ReadLegacy(IReadObjectContext * context, IStream * stream)
...
+    else
+    {
+        // When not in select by track type mode, add boosters if the track type is eligible
+        for (int i = 0; i < 3; i++) {
+            if (_legacyType.ride_type[i] == RIDE_TYPE_LOOPING_ROLLER_COASTER ||
+                _legacyType.ride_type[i] == RIDE_TYPE_CORKSCREW_ROLLER_COASTER ||
+                _legacyType.ride_type[i] == RIDE_TYPE_TWISTER_ROLLER_COASTER ||
+                _legacyType.ride_type[i] == RIDE_TYPE_VERTICAL_DROP_ROLLER_COASTER ||
+                _legacyType.ride_type[i] == RIDE_TYPE_GIGA_COASTER ||
+                _legacyType.ride_type[i] == RIDE_TYPE_JUNIOR_ROLLER_COASTER) {
```
Comment (IntelOrca) [inline src/object/RideObject.cpp]: Braces go on separate lines. Line 79 also suffers from this, so feel free to fix that line too.

File: distribution/changelog.txt [hunk 3]
```diff
@@ -3,6 +3,7 @@
 - Feature: Allow entertainers' costume changes even in absence of required scenery
 - Feature: FreeBSD support
 - Feature: Allow loading of parks from URLs
+- Feature: Re-add boosters [#4963]
```
Comment (IntelOrca) [inline distribution/changelog.txt]: Rather than saying "Re-add", you could say, Introduce boosters from RCT1.

**Comment by X123M3-256** [comment]
This is a great addition but the booster is very slow - it requires a disproportionately long launch track to get up to speed, even with short trains. I couldn't get the train to clear a 30m hill even with a very long launch run. Could this be increased, or are you trying to recreate the behaviour in RCTC?

**Comment by YoloSweggLord** [comment]
One annoyance I have with this is that, after building a booster section of track, the next track piece defaults to a straight section of track, requiring you to re-select the booster track piece from the special track pieces list. Would it be possible to change this behavior to how it is with brakes, where after placing a brake piece, the next track piece is automatically selected as a brake piece? If that's a bit confusing I'll try to make a .gif explaining what I mean.

**Comment by Gymnasiast** [comment]
@X123M3-256 I'm trying to recreate the behaviour of RCT1. RCTC might have the same behaviour, but I can't verify that because I don't own it (or a compatible device, for that matter).

@YoloSweggLord I understand exactly what you mean. Could you open an Issue if there isn't one already?

**Comment by YoloSweggLord** [comment]
Issue #4975 already exists concerning that.

**Pull request merged**